packages/dds/tree/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -469,6 +427,51 @@ export function tryExtendMark(
     return false;
 }
 
+function tryMergeMoves(
+    end: MoveEnd,
+    left: MoveMark<unknown>,
+    right: MoveMark<unknown>,
+    revision: RevisionTag | undefined,
+    moveEffects: MoveEffectTable<unknown>,
+    recordMerges: boolean,
+): boolean {
+    if (left.conflictsWith !== right.conflictsWith) {
+        return false;
+    }
+    const rev = left.revision ?? revision;
+    const oppEnd = end === MoveEnd.Source ? MoveEnd.Dest : MoveEnd.Source;
+    const prevMergeId = getMoveEffect(moveEffects, oppEnd, rev, left.id).mergeRight;
+    if (prevMergeId !== undefined && prevMergeId !== right.id) {
+        makeMergeable(moveEffects, oppEnd, rev, prevMergeId, right.id);
+    } else {
+        makeMergeable(moveEffects, oppEnd, rev, left.id, right.id);
+    }
+
+    if (getMoveEffect(moveEffects, end, rev, left.id).mergeRight === right.id) {
```

Comment:
Would it be possible to pull out `getMoveEffect(moveEffects, end, rev, left.id)` into a local and reuse it instead of calling `getOrAddEffect(moveEffects, end, rev, left.id)` below?